typescript/sdk/src/deploy/EvmModuleDeployer.ts

```diff
@@ -74,17 +92,37 @@ export class EvmModuleDeployer<Factories extends HyperlaneFactories> {
       await this.multiProvider.handleTx(chain, initTx);
     }
 
-    const verificationInput = getContractVerificationInput({
-      name: contractName,
-      contract,
-      bytecode: factory.bytecode,
-      expectedimplementation: implementationAddress,
-    });
+    const { protocol } = this.multiProvider.getChainMetadata(chain);
+    const isZKSyncChain = protocol === ProtocolType.ZKSync;
+
+    let verificationInput: ContractVerificationInput;
+    if (isZKSyncChain) {
+      verificationInput = await getContractVerificationInputForZKSync({
+        name: contractName,
+        contract,
+        constructorArgs: constructorArgs,
+        artifact: artifact,
+        expectedimplementation: implementationAddress,
+      });
+    } else {
+      verificationInput = getContractVerificationInput({
+        name: contractName,
+        contract,
+        bytecode: factory.bytecode,
+        expectedimplementation: implementationAddress,
+      });
+    }
+
     this.addVerificationArtifacts({ chain, artifacts: [verificationInput] });
 
     // try verifying contract
     try {
-      await this.contractVerifier?.verifyContract(chain, verificationInput);
+      if (isZKSyncChain) {
+        const verifier = new ZKSyncContractVerifier(this.multiProvider);
```

Comment:
set it up once at the start when we set up the base contractVerifier? that we we don't need to create a new one for every verifyContract call